### PR TITLE
chore: feat/refactor-auth-provider-to-cookies

### PR DESCRIPTION
**Pull request by shahzaib-deriv**
This PR focuses on moving the usage of localStorage to Cookies to be syncable without using any localstorage event listeners

### CODE REVIEW

**Review by thisyahlen-deriv — APPROVED** [review]

File: src/context/auth-context.tsx [hunk 1]
```diff
@@ -19,55 +26,59 @@ type AuthDataProviderProps = {
...
+
+    const switchAccount = useCallback(
+        (loginId: string) => {
+            const accountsList: AccountsList = JSON.parse(Cookies.get('accountsList') ?? '[]');
+
+            const token = accountsList.find(account => account.loginid === loginId)?.token;
+            if (!token) return;
+            authorizeAccount(token);
+
+            Cookies.set('authToken', token);
```
Comment (jim-deriv) [inline src/context/auth-context.tsx:44]: @shahzaib-deriv, is everyone okay with this change ? `'client.active_loginid'` to `'authToken'` is there a thread regarding this? (I personally prefer this too actually, but...)
```suggestion
            Cookies.set('authToken', token);
```

**Review by jim-deriv — APPROVED** [review]

File: src/context/auth-context.tsx (hunk 1 above)
Comment (shahzaib-deriv) [reply src/context/auth-context.tsx:44]: @jim-deriv this is how we want to go forward, we don't need to rely on storage anymore. Had a discussion with @thisyahlen-deriv and we implemented this in Tradershub v2.

Cookies are also accessible on domains without the need to syncLocalStorage script

File: src/context/auth-context.tsx [hunk 2]
```diff
@@ -19,55 +26,59 @@ type AuthDataProviderProps = {
...
+
+    const { data, mutate, isSuccess, isPending } = useAuthorize();
+
+    const authorizeAccount = useCallback((token?: string) => {
+        if (token) mutate({ authorize: token });
+    }, []);
+
+    const switchAccount = useCallback(
+        (loginId: string) => {
+            const accountsList: AccountsList = JSON.parse(Cookies.get('accountsList') ?? '[]');
```
[the remark this answers is not in the document]
Comment (balakrishna-deriv) [reply src/context/auth-context.tsx:38]: yes, @shahzaib-deriv this is the client-side cookie. But we should think about including this to the deriv-utils package - https://github.com/deriv-com/deriv-utils
This way we can abstract the auth cookies read and write and don't have to implement it in the other packages.

**Comment by thisyahlen-deriv** [comment]
TODO
A method will be added to sync localStorage with legacy code.

**Review by thisyahlen-deriv — APPROVED** [review]

**Comment by shahzaib-deriv** [comment]
LGTM

**Review by matin-deriv — APPROVED** [review]

**Pull request merged**